### PR TITLE
refactor(run-protocol): convert runStakeKit to vobj

### LINKED ISSUE #4534: runStake: use virtual objects for high-cardinality state
part of #4511
There's a documentation task in #4518 .

### PR DESCRIPTION
**Pull request by turadg**
closes: #4534

## Description

RunStakeKit is the high cardinality object of RUNstake. This converts it to being a virtual object, completing #4534.

It also fixes part of the typedef for PlusContext. I'll do a merge commit to separate them in master.

### Security Considerations

State can be paged out and in. Relies on security of vobj system.

### Documentation Considerations

--

### Testing Considerations

Refactor. Existing tests suffice. Durability out of scope.

### CODE REVIEW

File: packages/run-protocol/src/runStake/runStakeKit.js [hunk 1]
```diff
@@ -128,6 +136,292 @@ const initState = (zcf, startSeat, manager) => {
...
+  /**
+   * @param {MethodContext} context
+   * @param {ZCFSeat} seat
+   */
+  getRunAllocated: ({ state }, seat) =>
+    seat.getAmountAllocated(KW.Debt, state.debtBrand),
+  /**
+   * @param {MethodContext} context
+   */
+  getCollateralAmount: ({ state, facets }) => {
```
Comment (dckc) [inline packages/run-protocol/src/runStake/runStakeKit.js:150]: What about:

```suggestion
  getCollateralAmount: ({ state, { helper }) => {
```

so we can leave `helper.getCollateralAllocated(...)` alone?

Comment (dckc) [reply packages/run-protocol/src/runStake/runStakeKit.js:150]: or `const { helper } = facets`?

**Review by dckc — APPROVED** [review]
I reviewed this somewhat lightly. Looks OK.
One minor suggestion; no need to act on it.

File: packages/run-protocol/src/runStake/runStakeKit.js (hunk 1 above)
Comment (turadg) [reply packages/run-protocol/src/runStake/runStakeKit.js:150]: I went with the latter. The double destructuring leads causes a circular type definition.

File: packages/run-protocol/src/runStake/params.js [hunk 2]
```diff
@@ -2,10 +2,12 @@
 // @jessie-check
 
 import { CONTRACT_ELECTORATE, ParamTypes } from '@agoric/governance';
+import { KW as AttKW } from './attestation.js';
```
Comment (dckc) [inline packages/run-protocol/src/runStake/params.js]: won't this cause the same problem?

Comment (turadg) [reply packages/run-protocol/src/runStake/params.js]: yeah would in next PR so fixed with [c73f6f2](https://github.com/Agoric/agoric-sdk/pull/5171/commits/c73f6f2d7e0e83c5ab94db47412df74f8e4a5e6c)

**Pull request merged**